graylog2-server/src/main/java/org/graylog2/indexer/indexset/profile/IndexFieldTypeProfileService.java

```diff
@@ -108,6 +113,16 @@ public IndexFieldTypeProfile save(final IndexFieldTypeProfile indexFieldTypeProf
         return super.save(indexFieldTypeProfile);
     }
 
+    @Override
+    public int delete(final String id) {
```

Comment:
Shouldn't this be the responsibility of the index set service? I'd maybe move it there and call it from here. It feels unexpected to modify the collection from a different service.

Reply:
Thanks, very good catch!